platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/analysis/PropertyAccessResolver.kt

```diff
@@ -73,8 +78,19 @@ class PropertyAccessResolverImpl(
     @Suppress("NestedBlockDepth")
     private
     fun AnalysisContext.doResolvePropertyAccessToObject(
-        propertyAccess: PropertyAccess
+        propertyAccess: PropertyAccess,
+        expectedType: DataTypeRef?
     ): ObjectOrigin? {
+        if (propertyAccess.receiver == null && expectedType != null && expectedType is DataTypeRef.Name) {
+            val dataType = schema.findTypeWithName(expectedType.fqName)
```

Comment:
Please use a `TypeRefContext` (`SchemaTypeRefContext` in particular) to get a `DataType` by `DataTypeRef`.